### PR TITLE
docs: use ts in examples, ensure package docs are up to date

### PR DESCRIPTION
**Pull request by achingbrain**
## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

Closes https://github.com/libp2p/js-libp2p/issues/1777

### CODE REVIEW

**Review by maschad — CHANGES_REQUESTED** [review]
We should wait until `doc-check` is re-enabled in https://github.com/libp2p/js-libp2p/pull/2250 before merging this to ensure they are up to date.

**Comment by achingbrain** [comment]
AFAIK #2250 is blocked on https://github.com/TypeStrong/ts-node/issues/2076 making it into a release so let's not wait for that here.

**Pull request merged**
